### DISCUSSION TITLE
CreateNPCFile overwriting existing files

### DISCUSSION THREAD

**Discussion by MrJoeMaster13** [EssentialsX/Essentials#4172] [Q&A] [answered]
Whenever I go into the server the balances of my towns/nations gets reset. The root of the problem is with Vault not recognizing that there is already an account for this, but I just want to cut to the chase and fix the Essentials' CreateNPCFile function to first test for any existing files before overwriting something with the same uuid. Below is the log report and the suggested if statement that I recommend be added to the com.earth2me.essentials.api.Economy.createNPCFile function. Can someone please help me compile a version of essentials that has this if statement and see if it fixes my issue?

![image](https://user-images.githubusercontent.com/84632447/119231300-2d4dfa80-baee-11eb-818b-ca248a8ef4c5.png)
![image](https://user-images.githubusercontent.com/84632447/119231310-38088f80-baee-11eb-9793-a9af7795b9ee.png)

**Comment by JRoy** [comment, marked as answer]
This has been fixed in f8f1facf43afdb960e87e07ec6e5e55701cacecc and will be further improved with #4188

**Reply by MrJoeMaster13** [reply]
It is still giving me the same issue after I updated essentials


![image](https://user-images.githubusercontent.com/84632447/120403467-4175d100-c312-11eb-9399-159cf0b31e28.png)

**Reply by MrJoeMaster13** [reply]
Is this the right version of essentials?:

![image](https://user-images.githubusercontent.com/84632447/120403684-b0532a00-c312-11eb-97d7-0d661d7ef37f.png)

**Reply by JRoy** [reply]
@MrJoeMaster13 can you share the entire output of `/ess version`

**Reply by MrJoeMaster13** [reply]
![image](https://user-images.githubusercontent.com/84632447/120404322-1be9c700-c314-11eb-9986-cc1aafa35633.png)

**Reply by MrJoeMaster13** [reply]
@JRoy Is this what you were looking for?
